centreon/src/Core/Dashboard/Playlist/Application/Repository/ReadPlaylistShareRepositoryInterface.php

```diff
@@ -33,7 +35,44 @@ interface ReadPlaylistShareRepositoryInterface
      * @param int $playlistId
      * @param ContactInterface $contact
      *
+     * @throws \Throwable
+     *
      * @return bool
      */
     public function existsAsEditor(int $playlistId, ContactInterface $contact): bool;
-}
\ No newline at end of file
+
+    /**
+     * Check if a user has a given playlist shared with him.
+     *
+     * @param int $playlistId
+     * @param ContactInterface $contact
+     *
+     * @throws \Throwable
+     *
+     * @return bool
+     */
+    public function exists(int $playlistId, ContactInterface $contact): bool;
```

Comment:
IMHO this is kind of confusing. The `exists` terminology indicates if a entry exists in the database for the ID provided. Here clearly it is not really the case, what is it checked is the "relation" (share) between a user and a playlist. Maybe this could be renamed ? So that if you ever want to check the existance of actual playlist(s) you can use `exists / exist `methods

Reply:
As discuss with @adr-mo there is plenty of exists methods in many repositories so, i'll keep this method name to be consistent with other implementations